lib/src/model/documentation.dart

```diff
@@ -3,12 +3,12 @@
 // BSD-style license that can be found in the LICENSE file.
 
 import 'package:dartdoc/src/markdown_processor.dart';
-import 'package:dartdoc/src/model/model.dart';
 import 'package:dartdoc/src/render/documentation_renderer.dart';
+import 'package:dartdoc/src/warnings.dart';
 import 'package:markdown/markdown.dart' as md;
 
 class Documentation {
-  final Canonicalization _element;
+  final Warnable _element;
```

Comment:
(Nothing actionable for this CL, just thoughts)
I think this is better. This reminds me that using the mixin as a type makes code so hard to read. I've come across this `Documentation` class so many times and had to figure out what class the `Warnable` was and where that class was instantiated.

Reply:
Yeah I agree. I think maybe using mixins as types is what led to some shoe-horning and complexity.